### DISCUSSION TITLE
[Docs]: Update validate_certs documentation to include CA path option

### DISCUSSION THREAD

**Discussion by thebaltimorefootstomper** [netbox-community/ansible_modules#803] [General]
### Change Type

Addition

### Area

Configuration parameters

### Proposed Changes

In the latest documentation I'm seeing for the Netbox Ansible Collection, at least for the VLAN module, the validate_certs 
 parameter in the documentation only includes the options yes or no. However, looking at the underlying Python requests module and from my testing, there is also an option to specify the path to the CA bundle file, which in my case was required to get the validate_certs option to work. With just yes, I would get the generic "Failed to establish connection to Netbox API" error.

**Comment by rodvand** [comment]
Have you checked out the `cert` option? You're also not proposing any changes so not sure what action you want us to take?

**Comment by thebaltimorefootstomper** [comment]
If I set the cert option instead to the CA path, and just validate_certs to yes, I still get the "Failed to establish connection to Netbox API" error. I thought the cert option was more for a client side certificate, not the CA path, please forgive if I'm misunderstanding. 
I guess the only proposed change I'm suggesting is to update the documentation to include the third optional value for the validate_certs parameter (in addition to yes or no), the CA bundle path (eg: /etc/pki/tls/cert.pem)

**Comment by sc68cal** [comment]
Changing this issue to a discussion since it appears to fit better.

**Comment by rodvand** [comment]
@thebaltimorefootstomper How did you set the CA cert then to get it working? Did you adjust some of the code manually?

**Reply by thebaltimorefootstomper** [reply]
I just set it in the Ansible playbook like :

validate_certs: "/etc/pki/tls/cert.pem"

I think this goes back to the underlying Python requests module thats being used?

From a snippet of the source of the Python requests module from adapters.py :

    def cert_verify(self, conn, url, verify, cert):
        """Verify a SSL certificate. This method should not be called from user
        code, and is only exposed for use when subclassing the
        :class:`HTTPAdapter <requests.adapters.HTTPAdapter>`.

        :param conn: The urllib3 connection object associated with the cert.
        :param url: The requested URL.
        :param verify: Either a boolean, in which case it controls whether we verify
            the server's TLS certificate, or a string, in which case it must be a path
            to a CA bundle to use
        :param cert: The SSL certificate to verify.

Key part I'm seeing is the verify parameter :

        :param verify: Either a boolean, in which case it controls whether we verify
            the server's TLS certificate, or a string, in which case it must be a path
            to a CA bundle to use

**Reply by thebaltimorefootstomper** [reply]
Here are my versions :

ansible            5.9.0
ansible-core       2.12.6
pynetbox           6.6.2
requests           2.28.0

netbox.netbox     3.7.1 

Python 3.8.11

Netbox 3.0.2

**Comment by juresaht2** [comment]
Just a reminder that this issue is still present.
